ucd-parse/src/arabic_shaping.rs

```diff
@@ -24,7 +24,114 @@ pub struct ArabicShaping {
     /// The "joining type" of this codepoint.
     pub joining_type: JoiningType,
     /// The "joining group" of this codepoint.
-    pub joining_group: String,
+    pub joining_group: JoiningGroup,
+}
+
+/// The Joining_Group field read from ArabicShaping.txt
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum JoiningGroup {
```

Comment:
Just looking at the values, it kind of feels like this should be marked as `#[non_exhaustive]`. I don't really have any confidence that this list won't expand in future versions of Unicode.

Honestly might be a good reason to just drop the enum altogether. Why did you make this an enum?

Reply:
> Honestly might be a good reason to just drop the enum altogether. Why did you make this an enum?

The addition of the `joining-group` sub-command was done to as part of our work to add support for Syriac shaping to Allsorts. We need to match on the joining group to do this:

https://github.com/yeslogic/allsorts/blob/ade4ef90dddecf63370d78b5ada58439e910faaf/src/scripts/syriac.rs

It was changed to an enum to be consistent with `joining_type`, which has been used just prior to implement Arabic shaping https://github.com/yeslogic/allsorts/blob/ade4ef90dddecf63370d78b5ada58439e910faaf/src/scripts/arabic.rs#L33-L67

I guess in the context of doing one then the other, it seems like an oversight that one was an enum and the other a string. In hindsight I should have realised this was a breaking change.

Reply:
How about we add `joining_group2` that is the enum and leave `joining_group` as a `String`? Then add a note to the (public) docs of `joining_group` that it will be dropped in the next semver bump of the crate.

What about the `non_exhaustive` bit?

Reply:
> How about we add `joining_group2` that is the enum and leave `joining_group` as a `String`? Then add a note to the (public) docs of `joining_group` that it will be dropped in the next semver bump of the crate.

Yes that sounds fine.

> What about the `non_exhaustive` bit?

Giving this some thought I'm a bit on the fence. `non_exhaustive` allows backwards compatibility when new variants are added but I'm not sure this is desirable in this this case. I.e. maybe the default behaviour of reporting unhandled variants is desirable so that you get a prompt to handle the new cases versus having them silently hit a catch all case. The scenario where new variants are added coincides with new Unicode releases so perhaps it's reasonable to require updating code when adding support for newer Unicode releases. Then again if you're just a downstream consumer of some unicode related thing maybe you don't care enough about those details and would just like your code to keep compiling... 🤔 

Perhaps it's worth noting that none of the other enums have it so far either and they also see new additions. E.g. I think it's beneficial that the addition of new scripts (which happens in most Unicode releases) prompts code to handle them.

Anyway I don't think I have strong enough feelings to advocate one way or another. Since you have more experience maintaining popular crates I'm happy to defer to what you think is best.